### PR TITLE
Add Tag column in images output

### LINKED ISSUE #1051: Inconsistent image listing, compared to podman
**Description**

Image listing is inconsistent between both programs, which makes this often a little bit confusing. In the forums, people often are already confused solely because of the coexistence of Podman and Buildah and their overlapping commands (`podman build` ⇔ `buildah run`). I think it would be helpful if such friction could be further reduced. Would thus be cool if image listing could be unified, as the content is the same except for the timestamp – if there is no reason for Buildah to keep using it the current way, of course.

Differences:

* The sort order is different.

* Podman splits Buildah’s IMAGE NAME into REPOSITORY and TAG.
    - Buildah could also do the split for better readability and parsing.

* The column order is different.
    - Buildah could move the IMAGE ID column after the image name/tag column(s), like Podman and Docker do.

* Buildah’s CREATED AT is absolute, whereas Podmans CREATED is relative.
    - Absolute date is more precise which is nice, so this could be kept.

* Buildah’s SIZE column has a space between number and unit.
    - Podman should also do this (Docker does, too).
    - (Didn’t want to create a separate issue for that in the Podman repo.)

* Btw, the invocation is different (`buildah images` ⇔ `podman image ls`)

* Podmans listing is way more compact.

**Describe the results you received:**

```
$ buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
414b5dbe710f         docker.io/library/debian:stable-slim                     Sep 4, 2018 23:21      58.4 MB
25d1a7471615         docker.io/library/julia:latest                           Sep 5, 2018 02:17      455 MB
9fe73ed91518         docker.io/library/rust:latest                            Sep 26, 2018 00:20     1.6 GB
9053d42a132c         localhost/gutenberg:latest                               Sep 29, 2018 20:09     112 MB
bc26f1ed35cf         docker.io/library/nginx:latest                           Sep 25, 2018 19:21     113 MB
24508ec0e667         docker.io/library/fedora:29                              Sep 7, 2018 21:20      268 MB
```

```
$ podman image ls 
REPOSITORY                 TAG           IMAGE ID       CREATED       SIZE
localhost/gutenberg        latest        9053d42a132c   2 hours ago   112MB
docker.io/library/rust     latest        9fe73ed91518   3 days ago    1.6GB
docker.io/library/nginx    latest        bc26f1ed35cf   4 days ago    113MB
docker.io/library/fedora   29            24508ec0e667   3 weeks ago   268MB
docker.io/library/julia    latest        25d1a7471615   3 weeks ago   455MB
docker.io/library/debian   stable-slim   414b5dbe710f   3 weeks ago   58.4MB
```

<details>
  <summary>System information</summary>
  
**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.4-1.dev.git0a7389c.fc29.x86_64
```

**Output of `buildah version`:**

```
buildah version 1.4-dev (image-spec 1.0.0, runtime-spec 1.0.0)
```

**Output of `podman version` if reporting a `podman build` issue:**

```
Version:       0.8.4
Go Version:    go1.11rc2
OS/Arch:       linux/amd64
```

**Output of `cat /etc/*release`:**

```
Fedora release 29 (Twenty Nine)
NAME=Fedora
VERSION="29.20180927.n.0 (Workstation Edition)"
ID=fedora
VERSION_ID=29
PLATFORM_ID="platform:f29"
PRETTY_NAME="Fedora 29.20180927.n.0 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:29"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=29
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=29
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
OSTREE_VERSION=29.20180927.n.0
Fedora release 29 (Twenty Nine)
Fedora release 29 (Twenty Nine)
```

**Output of `uname -a`:**

```
Linux khazad-dum 4.18.9-300.fc29.x86_64 #1 SMP Thu Sep 20 02:32:53 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"
```

</details>

### PR DESCRIPTION
**Pull request by QiWang19**
This can fix #1051 
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by QiWang19** [comment]
why buildah uses hard code in the output format? 
```fmt.Printf("%-20.12s %-56s", params.ID, params.Name)```

**Comment by rhatdan** [comment]
We are matching the output of Docker.

**Comment by QiWang19** [comment]
After this patch
```
IMAGE NAME                                               IMAGE TAG            IMAGE ID             CREATED AT             SIZE
k8s.gcr.io/pause                                         3.1                  da86e6ba6ca1         Dec 20, 2017 16:30     747 KB
quay.io/crio/redis                                       alpine               98bd7cfc43b8         Mar 27, 2018 14:50     28.1 MB
docker.io/library/golang                                 1.11                 d817ad5b9beb         Nov 2, 2018 19:50      795 MB
<none>                                                   <none>               2f35638c7403         Nov 8, 2018 14:50      981 MB

```

**Comment by rhatdan** [comment]
LGTM

File: imagebuildah/util.go [hunk 1]
```diff
@@ -111,3 +111,23 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
...
+
+// ReposToMap parses the specified repotags and returns a map with repositories
+// as keys and the corresponding arrays of tags as values.
+func ReposToMap(repotags []string) map[string][]string {
+	// map format is repo -> tag
+	repos := make(map[string][]string)
+	for _, repo := range repotags {
+		var repository, tag string
+		if len(repo) > 0 {
+			li := strings.LastIndex(repo, ":")
```
Comment (rhatdan) [inline imagebuildah/util.go]: What happens if repo does not have a ":"?

Comment (QiWang19) [reply imagebuildah/util.go]: use `<none>` as a tag? in what situation the `repo` without ":" will be fed to this function, the default tag it has is `:latest`?
```go
        var repository, tag string
        if strings.Contains(repo, ":") {
            li := strings.LastIndex(repo, ":")
            repository = repo[0:li]
            tag = repo[li+1:]
        }
        if !strings.Contains(repo, ":") && len(repo) > 0 {
            repository = repo
            tag = "<none>"
        }
        repos[repository] = append(repos[repository], tag)
```

Comment (rhatdan) [reply imagebuildah/util.go]:
```
        var repository, tag string
        if strings.Contains(repo, ":") {
            li := strings.LastIndex(repo, ":")
            repository = repo[0:li]
            tag = repo[li+1:]
        } else if len(repo) > 0 {
            repository = repo
            tag = "<none>"
        }
        repos[repository] = append(repos[repository], tag)
```

But should we return an error if len(repo) == ""?

**Comment by TomSweeneyRedHat** [comment]
The `buildah images` man page has some example output that ought to be touched up.

File: cmd/buildah/images.go [hunk 2]
```diff
@@ -270,48 +272,52 @@ func outputImages(ctx context.Context, images []storage.Image, store storage.Sto
...
-				data, err2 := json.MarshalIndent(JSONImage, "", "    ")
-				if err2 != nil {
-					return err2
+				if !matchesFilter(ctx, image, store, name, filters) {
+					continue
+				}
+				if opts.quiet {
+					fmt.Printf("%-64s\n", image.ID)
+					// We only want to print each id once
+					break
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/images.go]: This break is breaking out of the new `for _, tag := range tags {` loop now, but we need to break out of the original for loop `for name, tags := range imagebuildah.ReposToMap(names) {` still.  Need to figure out how to break and go to the next name and stop processing on the tags when --quiet is selected.

**Pull request closed without merging**